docs/proposals-accepted/202209-receive-tenant-external-labels.md

```diff
@@ -0,0 +1,72 @@
+---
+type: proposal
+title: Allow specifying tenant-specific external labels in Receivers
+status: accepted
+owner: haanhvu
+menu: proposals-accepted
+---
+
+## 1 Related links/tickets
+
+https://github.com/thanos-io/thanos/issues/5434
+
+## 2 Why
+
+We would like to select and query tenant blocks from Receivers in an arbitrary manner.
+
+## 3 Pitfalls of the current solution
+
+Currently, we can only add external labels to Receiver itself, not to each tenant in the Receiver. There’s currently no way to select and query tenant blocks in an arbitrary manner.
+
+## 4 Goals
+
+* Allow users to add arbitrary tenants’ external labels in the easiest way possible
+* Allow users to change arbitrary tenants’ external labels in the easiest way possible
+* Changes in tenants’ external labels are handled correctly
+* Backward compatibility (e.g., with Receiver’s external labels) is assured
+* Tenants’ external labels are handled separately in RouterIngestor, RouterOnly, and IngestorOnly modes
+
+## 5 Non-goals
+
+* Logically split RouterOnly and IngestorOnly modes (Issue [#5643](https://github.com/thanos-io/thanos/issues/5643)): If RouterOnly and IngestorOnly modes are logically split, implementing tenants’ external labels in RouterOnly and IngestorOnly modes would be less challenging. However, the split could be tricky and time-consuming, which might distract us from implementing tenants’ external labels. So we will not do it now.
+
+## 6 Audience
+
+Users who need to have cross-tenant information and do cross-tenant activities
```

Comment:
This isn't really cross-tenancy. Maybe something like this would be better?
```suggestion
Users who are admin personas and need to perform admin operations on Thanos for multiple tenants
```